tls/s2n_connection.c

```diff
@@ -659,6 +656,13 @@ int s2n_connection_wipe(struct s2n_connection *conn)
         conn->actual_protocol_version = s2n_highest_protocol_version;
     }
 
+    /* Initialize remaining values */
```

Comment:
Nit: I think this comment could instead describe why the initialization of these values is deferred to this part of the function. It's not very clear to me why we've moved them to the end. Do they have to be last for some reason?

Reply:
Huh maybe we should make the s2n_connection_zero more obvious / put a big comment on it saying that all initialization must occur after that point.

Reply:
`s2n_connection_new`: allocates memory for the connection (calls to `_alloc`) and some init (`stuffers`)

`s2n_connection_wipe`: wipe conn state and allow for reuse.
- save and restore some state across call to `s2n_connection_zero` (POSIX_CHECKED_MEMCPY)
- 'Initialize remaining values' (these settings dont need to live across connection and can just be set). so `blinding` flag, session_ticket_status, and cookie_stuffer value.



- This PR removes some [redundant init](https://github.com/aws/s2n-tls/pull/3282/files#diff-d475fb37d8e79eacc8fe159d7667aa423e47e78f650b1f8a5fc28940384c73afL98-L114) because `s2n_connection_zero` does this
- It calls [`s2n_connection_wipe` last](https://github.com/aws/s2n-tls/pull/3282/files#diff-d475fb37d8e79eacc8fe159d7667aa423e47e78f650b1f8a5fc28940384c73afR152) which ensure that init is done correctly in wipe
- And properly [frees](https://github.com/aws/s2n-tls/pull/3282/files#diff-d475fb37d8e79eacc8fe159d7667aa423e47e78f650b1f8a5fc28940384c73afR575) and [inits](https://github.com/aws/s2n-tls/pull/3282/files#diff-d475fb37d8e79eacc8fe159d7667aa423e47e78f650b1f8a5fc28940384c73afR664) `conn->cookie_stuffer`


---
What could be improved is to move all init from s2n_connection_new to s2n_connection_wipe but I wanted to limit the changes in this PR and address cookie_stuffer issue.